functions/wait/waitFor.go

```diff
@@ -5,44 +5,42 @@ import (
 
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
 	framework "github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/tfp-automation/config"
 	waitAction "github.com/rancher/tfp-automation/functions/wait/action"
 	waitState "github.com/rancher/tfp-automation/functions/wait/state"
-	"github.com/rancher/tfp-automation/config"
 )
 
-func WaitFor(t *testing.T, client *rancher.Client, clusterID string, action string) {
+const (
+	active            = "active"
+	provisioning      = "provisioning"
+	kubernetesUpgrade = "kubernetes-upgrade"
+	scaling           = "scaling"
+)
+
+// ExecuteAction is a function that will execute the specified module action.
+func ExecuteAction(t *testing.T, client *rancher.Client, clusterID, action string, clusterConfig *config.TerratestConfig) {
 	terraformConfig := new(config.TerraformConfig)
 	framework.LoadConfig("terraform", terraformConfig)
 
 	module := terraformConfig.Module
 
-	if module == "aks" || module == "eks" || module == "ec2_k3s" || module == "ec2_rke1" || module == "ec2_rke2" || module == "linode_k3s" || module == "linode_rke1" || module == "linode_rke2" {
-		if module != "eks" && !((module == "ec2_rke1" || module == "linode_rke1") && action == "kubernetes-upgrade") {
-			waitState.WaitingOrUpdating(t, client, clusterID)
+	if action == provisioning {
+		waitState.ActiveCluster(t, client, clusterID)
+		waitState.ActiveNodes(t, client, clusterID)
+	} else {
+		if action == "scale-up" {
```

Comment:
looking into each of the functions called, am I misreading? I don't think any of these are actually taking an action. rather, they are waiting for an action to complete. In the rancher framework, we typically just omit the 'action' / replace it. This is really a helper to wait for the cluster / resource to get to a certain state. Or in other words, you are validating that the cluster/resources are in a certain state. I think this file should be moved to the new extensions/verify.go file.

I think these if statements should be broken out into helpers. This will help new functions be easily added in the future, instead of being in an if block.

I'd recommend the following helpers:
`IsProvisioningComplete`
`IsScalingComplete`

Reply:
Resolved